src/GitHub.Api/Application/ApplicationManagerBase.cs

```diff
@@ -51,47 +51,70 @@ public void Run()
                     {
-                        Logger.Trace("Existing Environment Path Original:{0} Updated:{1}", Environment.Path, path);
-                        Environment.Path = path?.Split(new[] { "\"" }, StringSplitOptions.None)[1];
+                        var getEnvPath = new SimpleProcessTask(token, "bash".ToNPath(), "-c \"/usr/libexec/path_helper\"")
+                                   .Configure(ProcessManager, dontSetupGit: true)
+                                   .Catch(e => true); // make sure this doesn't throw if the task fails
+                        var path = getEnvPath.RunWithReturn(true);
+                        if (getEnvPath.Successful)
+                        {
+                            Logger.Trace("Existing Environment Path Original:{0} Updated:{1}", Environment.Path, path);
+                            Environment.Path = path?.Split(new[] { "\"" }, StringSplitOptions.None)[1];
+                        }
                     }
-                }
-
-                GitInstallationState state = new GitInstallationState();
-                var runInstallers = firstRun;
 
-                if (!runInstallers)
-                {
+                    bool skipInstallers = false;
                     state = SystemSettings.Get<GitInstallationState>(Constants.GitInstallationState) ?? state;
                     var now = DateTimeOffset.Now;
-                    runInstallers = now.Date != state.GitLastCheckTime.Date || (!(state.GitIsValid && state.GitLfsIsValid));
-                }
+                    if (now.Date == state.GitLastCheckTime.Date && state.GitIsValid && state.GitLfsIsValid)
+                    {
+                        // just check if the git/git lfs version is what we need
+                        if (firstRun)
+                        {
+                            var version = new GitVersionTask(token)
+                                .Configure(ProcessManager, state.GitExecutablePath, dontSetupGit: true)
+                                .Catch(e => true)
+                                .RunWithReturn(true);
+                            state.GitIsValid = version >= Constants.MinimumGitVersion;
+                            if (state.GitIsValid)
+                            {
+                                version = new GitLfsVersionTask(token)
+                                .Configure(ProcessManager, state.GitLfsExecutablePath, dontSetupGit: true)
+                                .Catch(e => true)
+                                .RunWithReturn(true);
+                                state.GitLfsIsValid = version >= Constants.MinimumGitLfsVersion;
+                            }
+                        }
+                    }
 
-                if (runInstallers)
-                {
-                    Environment.OctorunScriptPath = new OctorunInstaller(Environment, TaskManager)
-                        .SetupOctorunIfNeeded();
+                    if (!skipInstallers)
```

Comment:
I think you mean to set the value of `skipInstallers` with the result in `state`.